### DISCUSSION TITLE
Build windows images with buildx in Kubernetes

### DISCUSSION THREAD

**Discussion by spyrosmouchlianitis** [docker/buildx#2202] [Q&A] [answered]
Hello everyone,

I'm currently in the process of migrating my self-hosted Azure agents to AKS. Given that these agents are utilized for Docker builds, and following Microsoft's recommendation to use buildx, I opted to create a builder with the Kubernetes driver.

The builder functions correctly for Linux builds, but I encounter issues when attempting a Windows build, specifically when a `RUN` command is present. According to [this article](https://lippertmarkus.com/2022/04/22/win-img-docker-hostprocess-k8s/), running Windows builds with `RUN` commands is not supported. However, considering the age of the article, I'm curious if there have been any updates or resolutions to this issue?

Thank you in advance.

**Comment by jedevc** [comment]
Buildkit is working on support for windows containers: https://github.com/moby/buildkit/issues/616

As far as I'm aware, `RUN` should work now (with some caveats)? cc @gabriel-samfira @TBBle, would it be possible to get a doc or something in buildkit that covers the current state of windows (since even I'm not 100% sure about it!)

**Reply by TBBle** [reply, marked as answer]
I think https://github.com/microsoft/Windows-Containers/issues/34 is the better place to cover the overall state of user-facing stuff, and includes more user-facing experiments and results.

For example, simply having BuildKit working itself will not deliver buildx builder integration, or even a particularly user-friendly flow in BuildKit. (Using BuildKit directly is already relatively non-user-friendly, particularly on Windows where we don't yet have a strong packaging ecosystem delivering the dependencies.)

And to use buildx builders that rely on BuildKit in a container is something we haven't even started doing more than [speculating about](https://github.com/microsoft/Windows-Containers/issues/34#issuecomment-1811519476). Although I note that the linked article from 2022 has exercised a few of the mechanisms that we will need, which is nice.

**Reply by spyrosmouchlianitis** [reply]
Great, thanks for the insights guys! I'll try to find some alternative solutions then.
